File: Iter.Api/Iter.Core/Models/Agency.cs
namespace Iter.Core.Models
{
    public class Agency
    {
        public Agency()
        {
            DateCreated = DateTime.Now;
            DateModified = DateTime.Now;
        }
        public Guid Id { get; set; }

        public string Name { get; set; }

        public Guid AddressId { get; set; }

        public Address Address { get; set; }

        public string ContactEmail { get; set; }

        public string ContactPhone { get; set; }

        public string Website { get; set; }

        public string LicenseNumber { get; set; }

        public bool IsActive { get; set; }

        public string LogoUrl { get; set; }

        public decimal Rating { get; set; }


        public bool IsDeleted { get; set; }

        public DateTime DateCreated { get; set; }

        public DateTime DateModified { get; set; }


        public List<User> Users{ get; set; }

        public List<Arrangement> Arrangements{ get; set; }
    }
}

File: Iter.Api/ui/lib/models/agency.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ui/models/image_model.dart';
import 'address.dart';

part 'agency.g.dart';

@JsonSerializable()
class Agency {
  final String id;
  final String name;
  final Address? address;
  final String contactEmail;
  final String contactPhone;
  final String website;
  final String licenseNumber;
  final ImageModel? logo;
  final bool isActive;
  final double rating;
  final bool isDeleted;
  final DateTime dateCreated;
  final DateTime dateModified;

  Agency({
    required this.id,
    required this.name,
    this.address,
    required this.contactEmail,
    required this.contactPhone,
    required this.website,
    required this.licenseNumber,
    required this.isActive,
    required this.logo,
    required this.rating,
    required this.isDeleted,
    required this.dateCreated,
    required this.dateModified,
  });

  factory Agency.fromJson(Map<String, dynamic> json) => _$AgencyFromJson(json);

  Map<String, dynamic> toJson() => _$AgencyToJson(this);
}